### PR TITLE
MGMT-8127: Webhook block unbind if Agent is installing

### PR DESCRIPTION
**Pull request by rollandf**
# Assisted Pull Request

## Description
Reject change in ClusterDeploymentName in Agent CR if Agent is currently installing.

## List all the issues related to this PR

- [x] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @isaacdorfman 
/cc @filanov 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by rollandf** [comment]
/hold
Need to verify e2e with operator

File: pkg/validating-webhooks/agentinstall/v1beta1/agent_admission_hook.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+// ValidatingResource is called by generic-admission-server on startup to register the returned REST resource through which the
+//                    webhook is accessed by the kube apiserver.
+// For example, generic-admission-server uses the data below to register the webhook on the REST resource "/apis/admission.agentinstall.openshift.io/v1/agentvalidators".
+//              When the kube apiserver calls this registered REST resource, the generic-admission-server calls the Validate() method below.
+func (a *AgentValidatingAdmissionHook) ValidatingResource() (plural schema.GroupVersionResource, singular string) {
+	log.WithFields(log.Fields{
+		"group":    agentAdmissionGroup,
+		"version":  agentAdmissionVersion,
+		"resource": "agentvvalidator",
```
Comment (filanov) [inline pkg/validating-webhooks/agentinstall/v1beta1/agent_admission_hook.go:42]:
```suggestion
		"resource": "agentvalidator",
```
Maybe it should be a const?

File: internal/controller/controllers/agentserviceconfig_controller.go [hunk 2]
```diff
@@ -1342,6 +1343,60 @@ func (r *AgentServiceConfigReconciler) newInfraEnvWebHook(ctx context.Context, l
...
+		{
+			Name:          "agentvalidators.admission.agentinstall.openshift.io",
+			FailurePolicy: &fp,
+			SideEffects:   &se,
+			AdmissionReviewVersions: []string{
+				"v1",
+			},
+			ClientConfig: admregv1.WebhookClientConfig{
+				Service: &admregv1.ServiceReference{
+					Namespace: "default",
```
[the remark this answers is not in the document]
Comment (rollandf) [reply internal/controller/controllers/agentserviceconfig_controller.go]: Done

File: pkg/validating-webhooks/agentinstall/v1beta1/agent_admission_hook.go (hunk 1 above)
Comment (rollandf) [reply pkg/validating-webhooks/agentinstall/v1beta1/agent_admission_hook.go]: Fixed typo.
I did not add a constant for it, only two occurrences.

**Comment by rollandf** [comment]
Verified E2E:
 - Start an installation
 - Try to Unbound the Agent
 
```
# oc -n assisted-installer edit agents.agent-install.openshift.io 0f0cd2d0-a52a-4d03-ba1b-e550c1eee21e 
error: agents.agent-install.openshift.io "0f0cd2d0-a52a-4d03-ba1b-e550c1eee21e" could not be patched: admission webhook "agentvalidators.admission.agentinstall.openshift.io" denied the request: Attempted to change Spec.ClusterDeploymentName which is immutable during Agent installation.
```

**Pull request merged**